### ISSUE TITLE
Remove AppSupport model

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#12315]
`AppSupport` is used to denormalize application support info, making it easier to find out what apps an add-on support from the `Addon` instance. It was used before in various parts of legacy frontend, when making sql queries listing add-ons compatible with the current app.

The new frontend doesn't rely on it, instead using the search API for all its "browse" queries, and therefore depending on elasticsearch, not sql, making that model obsolete.

Let's remove it. A few things still depend on it but it should be possible to work around that.

**Comment by diox** [comment]
I made a branch and it almost works, there is some weirdness to figure out with the language tools viewset/tests. https://github.com/mozilla/addons-server/compare/master...diox:remove-appsupport?expand=1

**Comment by diox** [comment]
I know we have been looking for potential good first python bugs, it might be more of a good *second* bug, but worth considering for someone that knows python, is familiar with unit testing and would be willing to spend some time debugging things.

**Comment by victorphoenix3** [comment]
@diox Hi! I'd like to work on this.

**Comment by diox** [comment]
@victorphoenix3 : keep in mind this is not a good first bug, but as I mentioned above it's rather a good *second* bug for someone that knows python, is familiar with unit testing and would be willing to spend some time debugging things.

To work on it: have a look at https://addons-server.readthedocs.io/en/latest/topics/install/docker.html for installing addons-server and then check out my branch linked in the comment above. There are a few unit tests that are failing and need some debugging to understand what's going on and fix.

**Comment by victorphoenix3** [comment]
@diox I have built addons-server and also created a local branch from your branch `remove-appsupport` on my machine.

**Comment by victorphoenix3** [comment]
You said in comments to your branch that its a single test failure. Would it be possible for you to please point out the test that is failing?

**Comment by diox** [comment]
https://travis-ci.org/diox/olympia/jobs/593594029 has the failing tests. Because the branch is a little outdated (*), the only one you should care about is `TestLanguageToolsView.test_memoize` in `src/olympia/addons/tests/test_views.py` (but it's worth re-testing the whole class or even the whole file to be sure). It's entirely possible that it's just a matter of updating the test itself (or the test data) - I stopped working on this before figuring out whether it was an actual bug with the branch or just an issue with the test.

(*) Once the test is fixed, the next task would be to update the branch with `master`, fixing conflicts along the way.

**Comment by victorphoenix3** [comment]
Hi @diox On `remove-Appsupport` branch the `addons-server_web_1` container does not start. So I am starting the containers on the `master` branch then changing my branch to `remove-Appsupport` to run the tests which are unsurprisingly failing.

But surprisingly the tests are also failing on the `master` branch. Have I done something wrong?

Traceback:

```
docker-compose exec web bash
olympia@a9970f9f7f60:/code$ pytest src/olympia/addons/tests/test_views.py::TestLanguageToolsView::test_memoize
/code/settings.py:143: UserWarning: Could not import local_settings module. Traceback (most recent call last):
  File "/code/settings.py", line 137, in <module>
    from local_settings import *  # noqa
ModuleNotFoundError: No module named 'local_settings'

  traceback.format_exc()))
{"Timestamp": 1582708575051513600, "Type": "raven.contrib.django.client.DjangoClient", "Logger": "http_app_addons", "Hostname": "a9970f9f7f60", "EnvVersion": "2.0", "Severity": 7, "Pid": 44, "Fields": {"uid": "", "remoteAddressChain": "", "msg": "Configuring Raven for host: None"}, "severity": 100}
{"Timestamp": 1582708575051980800, "Type": "raven.contrib.django.client.DjangoClient", "Logger": "http_app_addons", "Hostname": "a9970f9f7f60", "EnvVersion": "2.0", "Severity": 6, "Pid": 44, "Fields": {"uid": "", "remoteAddressChain": "", "msg": "Raven is not configured (logging is disabled). Please see the documentation for more information."}, "severity": 200}
{"Timestamp": 1582708575091320832, "Type": "raven.base.Client", "Logger": "http_app_addons", "Hostname": "a9970f9f7f60", "EnvVersion": "2.0", "Severity": 7, "Pid": 44, "Fields": {"uid": "", "remoteAddressChain": "", "msg": "Configuring Raven for host: None"}, "severity": 100}
{"Timestamp": 1582708575091561984, "Type": "raven.base.Client", "Logger": "http_app_addons", "Hostname": "a9970f9f7f60", "EnvVersion": "2.0", "Severity": 6, "Pid": 44, "Fields": {"uid": "", "remoteAddressChain": "", "msg": "Raven is not configured (logging is disabled). Please see the documentation for more information."}, "severity": 200}
============================================================================= test session starts ==============================================================================
platform linux -- Python 3.6.9, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/local/bin/python3
cachedir: .pytest_cache
django: settings: settings_test (from ini)
rootdir: /code, inifile: setup.cfg
plugins: forked-1.1.3, xdist-1.31.0, celery-4.4.0, django-3.8.0
collected 1 item                                                                                                                                                               

src/olympia/addons/tests/test_views.py::TestLanguageToolsView::test_memoize Using existing test database for alias 'default'...
ERROR

==================================================================================== ERRORS ====================================================================================
_____________________________________________________________ ERROR at setup of TestLanguageToolsView.test_memoize _____________________________________________________________
/deps/lib/python3.6/site-packages/pytest_django/plugin.py:524: in _django_setup_unittest
    request.getfixturevalue("django_db_setup")
/deps/lib/python3.6/site-packages/pytest_django/fixtures.py:108: in django_db_setup
    **setup_databases_args
/deps/lib/python3.6/site-packages/django/test/utils.py:174: in setup_databases
    serialize=connection.settings_dict.get('TEST', {}).get('SERIALIZE', True),
/deps/lib/python3.6/site-packages/django/db/backends/base/creation.py:72: in create_test_db
    run_syncdb=True,
/deps/lib/python3.6/site-packages/django/core/management/__init__.py:148: in call_command
    return command.execute(*args, **defaults)
/deps/lib/python3.6/site-packages/django/core/management/base.py:364: in execute
    output = self.handle(*args, **options)
/deps/lib/python3.6/site-packages/django/core/management/base.py:83: in wrapped
    res = handle_func(*args, **kwargs)
/deps/lib/python3.6/site-packages/django/core/management/commands/migrate.py:90: in handle
    executor.loader.check_consistent_history(connection)
/deps/lib/python3.6/site-packages/django/db/migrations/loader.py:299: in check_consistent_history
    connection.alias,
E   django.db.migrations.exceptions.InconsistentMigrationHistory: Migration waffle.0001_initial is applied before its dependency users.0001_initial on database 'default'.
```

**Comment by victorphoenix3** [comment]
Also the test failures on the two branches are different, on the `remove-Appsupport` branch I am getting assertion errors while on `master` I am getting migration errors.

**Comment by diox** [comment]
You shouldn't get any migration errors. After you switch branches, make sure to run `make update_docker` and if you're still seeing this, the first time after you switch branches, try passing `--create-db` to the test command.

**Comment by diox** [comment]
Still hoping for this to be done at some point to reduce complexity in our code.

**Comment by Sparsh-Bansal** [comment]
I would like to work on this issue.

**Comment by diox** [comment]
Feel free to give it a shot, you can use my old branch in https://github.com/mozilla/addons-server/compare/master...diox:remove-appsupport?expand=1 as a starting point - though it needs some heavy rebasing. The key thing will be to figure out the test failures though, and I'm afraid you're on your own for that.

**Comment by ioanarusiczki** [comment]
@eviljeff Please add some STR/ the areas that should be affected by this code removal and I'll take a look. 
Thanks!

**Comment by eviljeff** [comment]
@ioanarusiczki the model was referenced in a few places but it was really only the language tools api where it was still used (https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#language-tools).   So check https://addons.mozilla.org/firefox/language-tools/ is correct.

@diox are there other use cases for the language-tools API? (Firefox? different versions of Firefox viewing AMO?)

**Comment by diox** [comment]
Can't really think of anything else that was left, no.

**Comment by AlexandraMoga** [comment]
I've verified the https://addons.allizom.org/firefox/language-tools/ page and the [/language-tools/](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#language-tools) API and I haven't found issues.

For language packs, I've made sure that the API serves the version matching the used appversion - for example, if the language pack has a version compatible with Firefox 83.0 and another one for 84.0, the detail page on AMO will display the version appropriate for the Firefox version used.

Incompatibility messages are also displayed as expected.

I'm marking this issue as verified fixed on stage.
